devtools/x/src/installer.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{cargo::Cargo, config::CargoConfig};
+use log::info;
+use std::process::Command;
+
+pub fn install_if_needed(name: &str, version: &str) -> bool {
+    //This is a bit of a hack, given Cargo wants the CargoConfig struct, but we shouldn't be using the nightly
+    //cargo's tooling to build our 3rd party cargo tools.  Looking to address that in a refactoring pr once all the needed
+    //components are in place -- perhaps by seperating xcontext in to constituent parts.
+    let config = CargoConfig {
+        toolchain: "stable".to_owned(),
+        flags: Option::None,
+    };
+    install_if_needed_config(&config, name, version)
+}
+
+pub fn install_if_needed_config(config: &CargoConfig, name: &str, version: &str) -> bool {
+    if !check_installed(name, version) {
+        info!("Installing {} {}", name, version);
+        let installers = Cargo::new(config, "install")
+            .arg(name)
+            .arg("--version")
+            .arg(version)
+            .run();
+        if installers.is_err() {
+            info!(
+                "Could not install {} {}, check x.toml to ensure tool exists and is not yanked.",
+                name, version
+            );
+        }
+        installers.is_ok()
+    } else {
+        true
+    }
+}
+
+pub fn check_installed(name: &str, version: &str) -> bool {
+    let result = Command::new(name).arg("--version").output();
+    let found = match result {
+        Ok(output) => format!("{} {}", name, version)
+            .eq(String::from_utf8_lossy(&output.stdout).trim()),
+        Err(_) => false,
+    };
+    info!(
+        "{} of version {} is{} installed",
+        name,
+        version,
+        if !found { " not" } else { "" }
+    );
```

Comment:
So I don't think we want this to print out every time anyone ever runs xbuild or xtest. So this should either be a lower level or only output in the error case

Reply:
It wont, it'll only print out if we check for it.   Which will occur only when needed.   So I'm okay with it printing out before use.

Reply:
What I'm saying is I don't want a messages saying "sccache is installed" every time I run  `cargo xbuild`, assuming xbuild uses sccache. It makes sense to print a message out the first time something is installed, but not every time the tool is checked for

Reply:
I'm invisioning the log line stating that sccache is installed at a certain version occurring right before the env vars are set and logged, and the build line is logged.   Ideally I want folks to know exactly what's going on underneath the covers when xbuild runs, and seeing the sccache log line makes sense to me.   Young has used the log line of what's being built in the past to split out tests in to different runs....  I just want people to be able to drop to the command line and do exactly the same thing as the build, so I don't get called in to debug someone's build if I can avoid it.